Nos/Views/LikeButton.swift

```diff
@@ -5,20 +5,21 @@
 //  Created by Matthew Lorentz on 4/21/23.
 //
 
+import Logger
 import SwiftUI
 
 struct LikeButton: View {
     
     var note: Event
-    var action: () async -> Void
     @FetchRequest private var likes: FetchedResults<Event>
-    @EnvironmentObject private var currentUser: CurrentUser
     /// We use this to give instant feedback when the button is tapped, even though the action it performs is async.
     @State private var tapped = false
+    @EnvironmentObject private var relayService: RelayService
+    @EnvironmentObject private var currentUser: CurrentUser
+    @Environment(\.managedObjectContext) private var viewContext
     
-    internal init(note: Event, action: @escaping () async -> Void) {
+    internal init(note: Event) {
```

Comment:
Refactored the LikeButton so that it does the like by itself in order to avoid duplicated code when reusing the button